app/api/files/jsRoutes.js

```diff
@@ -132,4 +132,79 @@ export default app => {
       res.json('ok');
     }
   );
+
+  const parseQueryProperty = (query, property) =>
+    query[property] ? JSON.parse(query[property]) : query[property];
+
+  const generateExportFileName = databaseName => `${databaseName}-${new Date().toISOString()}.csv`;
+
+  app.get(
+    '/api/export',
+
+    validation.validateRequest(
+      Joi.object().keys({
+        filters: Joi.string(),
+        types: Joi.string(),
+        _types: Joi.string(),
+        fields: Joi.string(),
+        allAggregations: Joi.string(),
+        userSelectedSorting: Joi.string(),
+        aggregations: Joi.string(),
+        order: Joi.string(),
+        sort: Joi.string(),
+        limit: Joi.string(),
+        searchTerm: Joi.string().allow(''),
+        includeUnpublished: Joi.any(),
+        treatAs: Joi.string(),
+        unpublished: Joi.any(),
+        select: Joi.array(),
+        ids: Joi.array().items(Joi.string()),
+      }),
+      'query'
+    ),
+
+    (req, res, next) => {
+      req.query.filters = parseQueryProperty(req.query, 'filters');
+      req.query.types = parseQueryProperty(req.query, 'types');
+      req.query.fields = parseQueryProperty(req.query, 'fields');
+      req.query.aggregations = parseQueryProperty(req.query, 'aggregations');
+      req.query.select = parseQueryProperty(req.query, 'select');
+      req.query.unpublished = parseQueryProperty(req.query, 'unpublished');
+      req.query.includeUnpublished = parseQueryProperty(req.query, 'includeUnpublished');
+      req.query.ids = parseQueryProperty(req.query, 'ids');
+
+      Promise.all([search.search(req.query, req.language, req.user), settings.get()]).then(
+        // eslint-disable-next-line camelcase
+        ([results, { dateFormat, site_name }]) => {
+          const exporter = new CSVExporter();
+          const temporalFilePath = temporalFilesPath(
+            generateFileName({ originalname: 'export.csv' })
+          );
+          const fileStream = fs.createWriteStream(temporalFilePath, { emitClose: true });
+
+          const exporterOptions = {
+            dateFormat,
+          };
+
+          exporter
+            .export(results, req.query.types, fileStream, exporterOptions)
+            .then(() => {
+              fileStream.end(() => {
+                res.download(temporalFilePath, generateExportFileName(site_name), err => {
+                  if (err) {
+                    return handleError(err);
+                  }
+
+                  return fs.unlinkSync(temporalFilePath);
```

Comment:
Even if this could be fast, it is still a blocking procedure.  While using this in tests may work, please us non-sync unlink here.  Also, fs.unlink returns undefined, so be sure this is the expected behavior for success download.